torch/functional.py

```diff
@@ -87,8 +86,28 @@ def split(tensor, split_size_or_sections, dim=0):
     # call here.
     return tensor.split(split_size_or_sections, dim)
 
+# equivalent to itertools.product(indices)
+def indices_product(indices):
```

Comment:
Is there a way to bypass this when JIT isn't on?

Reply:
it's possible, but why exactly ? there should be a high bar for introducing divergent behavior for JIT & Python. 
This function produces the exact same values as `product`. When I ran 
```
import time
now = time.perf_counter()
for i in range(100):
    for shape in ((3, 3), (5, 3, 3), (7, 3, 5, 5), (7, 5, 3, 3, 3)):
    a = torch.randn(*shape, dtype=dtype, device=device)
    a_lu, p = torch.lu(a, pivot=pivot)
    p_ref, l_ref, u_ref = torch.lu_unpack(a_lu, p)
print(time.perf_counter() - now)
```
I consistently got 6.5 seconds for this implementation, and 7 seconds for the itertools.product implementation. So it also seems to be (somewhat) faster.

Reply:
I thought there might be a performance issue, but evidently not. Thank you for clarifying.